File: backend/world/models/state.py
from typing import List
from pydantic import BaseModel, Field
from world.models.entities import (
    Plant, Building, Zone, Sensor, Worker, Vehicle, Camera, Equipment,
    Pipeline, Valve, Permit, Hazard, EmergencyExit, WeatherState
)
from utils.datetime import format_iso, utc_now

class PlantState(BaseModel):
    """The root state object representing the entire digital twin of the industrial plant."""
    current_time: str = Field(default_factory=lambda: format_iso(utc_now()))
    weather: WeatherState
    
    plant: Plant
    buildings: List[Building] = Field(default_factory=list)
    zones: List[Zone] = Field(default_factory=list)
    workers: List[Worker] = Field(default_factory=list)
    sensors: List[Sensor] = Field(default_factory=list)
    equipment: List[Equipment] = Field(default_factory=list)
    pipelines: List[Pipeline] = Field(default_factory=list)
    valves: List[Valve] = Field(default_factory=list)
    vehicles: List[Vehicle] = Field(default_factory=list)
    hazards: List[Hazard] = Field(default_factory=list)
    permits: List[Permit] = Field(default_factory=list)
    cameras: List[Camera] = Field(default_factory=list)
    emergency_exits: List[EmergencyExit] = Field(default_factory=list)

File: services/go/internal/world/state.go
package world

import "time"

type PlantState struct {
	Version   int          `json:"version"`
	Timestamp time.Time    `json:"timestamp"`
	Plant     Plant        `json:"plant"`
	Buildings []Building   `json:"buildings"`
	Zones     []Zone       `json:"zones"`
	Sensors   []Sensor     `json:"sensors"`
	Workers   []Worker     `json:"workers"`
	Vehicles  []Vehicle    `json:"vehicles"`
	Cameras   []Camera     `json:"cameras"`
	Equipment []Equipment  `json:"equipment"`
	Pipelines []Pipeline   `json:"pipelines"`
	Valves    []Valve      `json:"valves"`
	Permits   []Permit     `json:"permits"`
	Hazards   []Hazard     `json:"hazards"`
	Exits     []EmergencyExit `json:"exits"`
	Weather   WeatherState `json:"weather"`
}
